### PR TITLE
chore: removes path monkey-patching, closes #2698

### LINKED ISSUE #2698: remove `path` monkey patch
Currently path is being monkey patched

https://github.com/orval-labs/orval/blob/c658fa3c40340fdc2dc155b2b7cabdc31809a16f/packages/core/src/utils/path.ts#L26-L44

While this works, it muddy the waters ☹️

Example:

```ts
// example A
import path from 'node:path';
```

```ts
// example B
import path from 'node:path';
import { upath } from '../utils';
```

Using `path` will do different things in example A and B.

I took a stab at it a while back, but it's non-obvious (to me) where the real path and the monkey patched path is being used so I focused on other things 😄

### PR DESCRIPTION
**Pull request by snebjorn**
Fix #2698

An effort was made to use `getRelativeImportPath` over other functions. This function focuses on creating correct esm import/export statement path strings. A lot more work is needed here.

## Summary by CodeRabbit

* **New Features**
  * Enhanced path utilities for consistent, cross-platform relative import generation.

* **Bug Fixes**
  * Normalized and simplified relative import paths across generated outputs.
  * More consistent path resolution when producing imports and re-exports.

* **Tests**
  * Added comprehensive tests covering relative import path generation and edge cases.

* **Chores**
  * Normalized snapshot and sample import paths to remove redundant segments.

### CODE REVIEW

File: packages/orval/src/write-specs.ts [hunk 1]
```diff
@@ -304,22 +310,23 @@ export async function writeSpecs(
         ? output.schemas
         : output.schemas.path;
       imports.push(
-        upath.relativeSafe(workspacePath, getFileInfo(schemasPath).dirname),
+        upath.getRelativeImportPath(
+          indexFile,
+          getFileInfo(schemasPath).dirname,
+        ),
```
[the remark this answers is not in the document]
Comment (snebjorn) [reply packages/orval/src/write-specs.ts:316]: Now I have all the context needed to verify this finding. Let me trace through the exact code path:
getFileInfo(schemasPath) calls isDirectory(target) on line 15 of file.ts
isDirectory is defined in assertion.ts line 21-23 as: return !path.extname(pathValue) — it checks for a file extension, not whether the path exists on disk
So getFileInfo is purely string-based and has no filesystem dependency
Let me trace the two representative cases:
schemasPath = '/project/src/schemas' (directory-style, no extension):
isDir = true → filePath = '/project/src/schemas/filename.ts' → dirname = '/project/src/schemas' ✓
schemasPath = '/project/src/schemas/index.ts' (file-style, has extension):
isDir = false → filePath = '/project/src/schemas/index.ts' → dirname = '/project/src/schemas' ✓
Both produce the correct dirname in every case, regardless of whether those paths exist on disk, because isDirectory never touches the filesystem.
The reported bug does not exist in the current code. The claim that getFileInfo(...).dirname "depends on the file existing" is incorrect — the implementation uses a pure extension check (!path.extname(pathValue)), making the result fully stable and deterministic.
No changes are needed. Applying the suggested fix (switching to upath.dirname(schemasPath)) would actually introduce a regression: for a directory-style path like /project/src/schemas, upath.dirname would return /project/src (the parent), while the current getFileInfo(...).dirname correctly returns /project/src/schemas itself.

File: packages/orval/src/utils/options.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { access } from 'node:fs/promises';
+import nodePath from 'node:path';
```
[the remark this answers is not in the document]
Comment (snebjorn) [reply packages/orval/src/utils/options.ts:2]: Now I have a complete picture. Let me trace the full data flow to verify the finding before making any changes.

**Key observations:**

1. **`NormalizedMutator.path`** (absolute) → resolved in `options.ts` via `nodePath.resolve(workspace, mutator.path)`.

2. **`generateMutator` in `mutator.ts`** converts that absolute path to a relative, Unix-normalized import path using `getRelativeImportPath(outputFile, mutator.path)` on line 13, and stores the result as `path` in the returned `GeneratorMutator` (line 92).

3. **`GeneratorMutator.path`** is therefore already a relative, Unix-normalized path (e.g. `./custom-instance` or `../custom-instance`). It is NOT absolute at this point.

4. **`generateMutatorImports` in `imports.ts` line 106** receives `GeneratorMutator[]`, not `NormalizedMutator[]`. The `mutator.path` there is the already-normalized relative path.

The proposed fix — `getRelativeImportPath(importerFilePath, mutator.path, includeExtension)` — **cannot be applied** because:
- `getRelativeImportPath` throws if either argument is not an absolute path (lines 95–102 of `path.ts`)
- `mutator.path` in `GeneratorMutator` is already relative, not absolute
- `generateMutatorImports` has no `importerFilePath` parameter and no access to one

The existing concatenation `${oneMore ? '../' : ''}${mutator.path}` produces paths like `.././custom-instance` (when `oneMore=true` and `mutator.path='./custom-instance'`). This is functionally identical to `../custom-instance` in every module resolver — the leading `./` is a no-op after `../`. There is no real bug here.

The cross-platform concern is already handled correctly: `getRelativeImportPath` in `mutator.ts` converts Windows backslashes to forward slashes before the result is stored in `GeneratorMutator.path`, so by the time `generateMutatorImports` in `imports.ts` sees it, it's already Unix-normalized.

**No fix is needed.** The suggestion as written would break the code — `getRelativeImportPath` enforces that both arguments are absolute paths (and throws otherwise), but at the `generateMutatorImports` call site, `mutator.path` is already a relative import specifier that was correctly normalized earlier in the pipeline by the `getImport` helper in `mutator.ts`.
